src/scripts/main.js

```diff
@@ -1,3 +1,52 @@
 'use strict';
 
 // write code here
+const headline = document.querySelector('thead');
+let lastClickedIndex = null;
+
+function getCleanedValue(row, index) {
+  const cell = row.querySelectorAll('td')[index];
+
+  if (!cell) {
+    return '';
+  }
+
+  const cleanedCell = cell.textContent.trim().replace(/[$,]/g, '');
+
+  return isNaN(cleanedCell) ? cleanedCell : parseInt(cleanedCell);
+}
+
+headline.addEventListener('click', (e) => {
+  const clickedElement = e.target.closest('th');
+
+  if (!clickedElement) {
+    return;
+  }
+
+  const columnIndex = clickedElement.cellIndex;
+  const tableRows = [...document.querySelectorAll('tbody tr')];
```

Comment:
You're querying the DOM for 'tbody tr' elements every time a click event is triggered. This can be inefficient if your table has many rows. Consider storing the table rows in a variable outside the event listener, and update it only when necessary (for example, when a row is added or removed). This way, you won't have to query the DOM every time a click event is triggered.